### ISSUE TITLE
Incorrect text shown in editable fields (e.g, M instead of 0)

### ISSUE THREAD

**Issue by banguero** [mozilla/pdf.js#7402]
Link to PDF file:
[fileWithProblem.pdf](https://github.com/mozilla/pdf.js/files/309449/fileWithProblem.pdf)

Configuration:
- Web browser and its version: Latest Chrome (reproes with any other)
- Operating system and its version: Windows 10
- PDF.js version: 1.5.292

Steps to reproduce the problem:
1. Go to https://mozilla.github.io/pdf.js/web/viewer.html 
2. Upload the file above to preview it
3. Notice that there are how 'Inspiration' and 'Strength' boxes show the incorrect values with PDFJS (M instead of 0)

What is the expected behavior? (add screenshot)
![image](https://cloud.githubusercontent.com/assets/1873511/15972599/55d014ec-2ef4-11e6-8f96-e567426a9877.png)

What went wrong? (add screenshot)
![image](https://cloud.githubusercontent.com/assets/1873511/15972646/958ef53a-2ef4-11e6-8e96-f9903e733798.png)

**Comment by Rob--W** [comment]
While loading that PDF, the following messages appear in the console:

```
Warning: fontRes not available (3 times)
Warning: Error during font loading: Font Helv is not available
Warning: Unknown command ">" (70 times)
```

**Comment by banguero** [comment]
@Rob--W - how can we fix it? is it a library issue?

**Comment by Rob--W** [comment]
I debugged a little bit and found that the issue does not happen when FontFace is disabled (`#disableFontFace=1`).

With `#pdfBug=all`, I can see that there are multiple fonts with the same (supposedly unique?) auto-generated name, "g_d0_fF0".

If I append `+ Math.random().toString(36).slice(3);` to the following line, then the PDF is displayed as expected (note that when I stepped through with the debugger, `fontRefIsDict` is true, so the `fontName` branch is taken).

https://github.com/mozilla/pdf.js/blob/a0a6e6d688bd43c0ba680025702d1b7228f4c89c/src/core/evaluator.js#L711-L714

I'll submit a PR to fix this, can we include your PDF file in our unit tests?
